java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -449,17 +449,50 @@ private boolean hasConstraintForDomain(LearnedConstraint lc, Class<? extends Con
       if (nodeTree.is(Tree.Kind.NEW_CLASS)) {
         return ImmutableList.of(location(parent, String.format("Constructor implies '%s'.", constraint.valueAsString())));
       }
-      String name = SyntaxTreeNameFinder.getName(nodeTree);
-      if (name == null) {
-        return ImmutableList.of();
-      }
       String msg;
-      if (ObjectConstraint.NULL == constraint && isNullCheck(nodeTree)) {
-        msg = String.format(IMPLIES_CAN_BE_NULL_MSG, name);
+      Tree reportTree = parent.programPoint.syntaxTree();
```

Comment:
This variable has same value as `nodeTree` already present. I think it would be better to use that one to test if the tree is final field tree and to directly return instead of nesting if-else constructs, otherwise it's becoming bit messy.